File: 239/main.rb
# O(n^2) solution
# current wrong

# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer[]}
def max_sliding_window(nums, k)
    return [] if nums.empty?
    result = []
    
    nums[0..-k].each_with_index do |num, i|
        temp = num
        nums[(i + 1)...(i + k)].each do |num|
            temp = num if num > temp
        end
        
        result << temp
    end
    
    result
end

File: 239/main.java
// O(n^2) solution

class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        int n = nums.length;
        if(nums == null || n <= 0) {
            return nums;
        }
        int[] result = new int[n - k + 1];
        
        for(int i = 0; i < n - k + 1; i++) {
            int temp = Integer.MIN_VALUE;
            for(int j = i; j < i + k; j++) {
                if(nums[j] > temp) {
                    temp = nums[j];
                }
            }
            
            result[i] = temp;
        }
        
        return result;
    }
}
